### DISCUSSION TITLE
Add filter supports for the reply function

### DISCUSSION THREAD

**Discussion by Trizze** [Chatterino/chatterino2#5449] [Ideas] [answered]
I have the term #spoiler filtered so that I don't see messages that contains it. But I still see it in replies unfortunately...
Which means I have to turn off the reply function in chatterino!

**Possible solutions:**
1. Censor message 1 with asterisks but keep the message 2 visible unless it contains a filtered word.
2. Censor both messages if either message contains a filtered word.

For me, I prefer the second solution but that's because I'm using it as a way to hide spoilers.

My filter:
![image](https://github.com/Chatterino/chatterino2/assets/31097605/5ad1c3a9-721a-4b35-b9a9-5cb21195007f)

How it looks like when people reply to the message:
![image](https://github.com/Chatterino/chatterino2/assets/31097605/54d354ee-a89e-44c5-869d-1e12efb411cf)
I don't see my original message, but when replied to; it's visible.

**Comment by Mm2PL** [comment, marked as answer]
The `message.content` variable only has the messages content, not the reply context. We should expose the reply thread content to the filters instead of adding that to `message.content`.

You can do an ignore on the word `#spoiler`. 

|Pattern|Regex|Case-sensitive|Block|Replacement|
|---|---|---|---|---|
|`.*#spoiler.*`|Yes|No|No|`#spoiler`|

This presents the message as `#spoiler` in replies. Nuking the message will show it in the replies, do not do that.

**Reply by Trizze** [reply]
Thanks for the quick reply.

I like the fact that the messages disappear with the filter so I decided to use a combination of my already existing filter and your method.

I don't see the initial spoiler message at all but if someone replies to it without a spoiler tag, it shows like this:
![image](https://github.com/Chatterino/chatterino2/assets/31097605/2230bc57-8d13-4345-83a6-b79848303f4f)

Thank you again for the help!
